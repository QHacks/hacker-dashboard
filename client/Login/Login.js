import React, { Component } from 'react';
import { Button, Form, Grid, Header, Image, Message, Segment } from 'semantic-ui-react';

export default class Login extends Component {
	render() {
		return (
			<p>Hello</p>
		);
	}
}
