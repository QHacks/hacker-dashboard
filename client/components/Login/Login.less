@import (less) "../../assets/semantic-ui/site/globals/site.variables";

@columnPadding: 50px;

.login-container {
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: center;
  height: 100vh;
  width: 100vw;
}

.login-form-wrapper {
  align-items: center;
  box-shadow: 3px 0 25px rgba(0,0,0,0.3);
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  padding: @columnPadding;
  width: 40%;
  z-index: 0
}

.qhacks-logo {
  height: 150px;
  margin-bottom: 20px;
}

.login-picture {
  align-items: center;
  background-color: @primaryColor;
  background-image: url("../../assets/img/students-at-hackathon.jpg");
  background-position: center;
  background-size: cover;
  color: white;
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  width: 60%;
}

.login-picture-overlay {
  align-items: center;
  background-color: rgba(0, 0, 0, 0.6);
  display: flex;
  height: 100%;
  justify-content: center;
  padding: @columnPadding;
  width: 100%;
}

@media (max-width: @tabletBreakpoint) {
  .login-container {
    height: 100%;
    padding: 30px;
  }

  .login-form-wrapper {
    box-shadow: none;
  }

  .login-picture {
    display: none;
  }

  .login-form-wrapper {
    width: 100%;
  }
}
