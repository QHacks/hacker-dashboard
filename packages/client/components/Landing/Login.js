import React, { Component } from "react";
import { Link } from "react-router-dom";
import { graphql } from "react-apollo";
import gql from "graphql-tag";
import axios from "axios";

import Landing from "./Landing";
import { SERVER_HOST } from "../../Client";
import ActionButton from "../ActionButton/ActionButton";
import StatusReport from "../StatusReport/StatusReport";

const ACCESS_TOKEN_STORAGE = "qhacksAccessToken";
const REFRESH_TOKEN_STORAGE = "qhacksRefreshToken";

const UPDATE_AUTHENTICATION_STATUS_MUTATION = gql`
  mutation UpdateAutheticationStatus($isAuthenticated: Boolean) {
    updateAuthenticationStatus(isAuthenticated: $isAuthenticated) @client
  }
`;

class Login extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: "",
      password: "",
      rememberMe: true
    };
  }

  async login() {
    const { email, password } = this.state;

    try {
      const response = await axios.post(`${SERVER_HOST}/oauth/session`, {
        email,
        password,
        grantType: "password"
      });

      localStorage.setItem(ACCESS_TOKEN_STORAGE, response.data.accessToken);
      localStorage.setItem(REFRESH_TOKEN_STORAGE, response.data.refreshToken);

      this.props.mutate({
        variables: {
          isAuthenticated: true
        }
      });
    } catch (err) {
      this.setState({ error: "Test Error" });
    }
  }

  render() {
    return (
      <Landing>
        <img
          src={"../../assets/img/qhacks-wordmark-colored.svg"}
          css={`
            max-height: 40px;
          `}
          alt="QHacks"
        />
        <p
          className="blurb"
          css={`
            margin-top: 24px;
            color: #8a929f;
          `}
        >
          Welcome back.
        </p>
        <p
          className="blurb"
          css={`
            margin-top: 12px;
            color: #8a929f;
          `}
        >
          Please login to your account.
        </p>
        <div
          css={`
            margin-top: 64px;
          `}
        >
          <input
            id="email"
            type="text"
            value={this.state.email}
            onChange={(e) => this.setState({ email: e.target.value })}
            placeholder="Enter your email address"
          />
          <input
            id="password"
            type="password"
            value={this.state.password}
            onChange={(e) => this.setState({ password: e.target.value })}
            placeholder="Enter your password"
          />
        </div>
        <div
          css={`
            margin: 40px 0;
            display: flex;
          `}
        >
          <div
            css={`
              flex-grow: 1;
            `}
          >
            <input
              type="checkbox"
              id="rememberMe"
              onChange={(e) => this.setState({ rememberMe: e.target.checked })}
            />
            <label htmlFor="rememberMe">Remember me</label>
          </div>
          <div
            css={`
              flex-grow: 1;
            `}
          >
            <Link className="landingLink" to="/forgot-password">
              Forgot password
            </Link>
          </div>
        </div>
        {this.state.error ? (
          <StatusReport type="danger" message={this.state.error} />
        ) : (
          ""
        )}
        <div>
          <ActionButton color="blue" onClick={() => this.login()}>
            Login
          </ActionButton>{" "}
          <ActionButton internal link="/qhacks-2019/apply">
            Apply
          </ActionButton>
        </div>
      </Landing>
    );
  }
}

export default graphql(UPDATE_AUTHENTICATION_STATUS_MUTATION)(Login);
