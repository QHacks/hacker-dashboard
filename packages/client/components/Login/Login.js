import { Redirect, Link } from "react-router-dom";
import React, { Component } from "react";
import MenuBar from "../MenuBar/index.js";
import * as colors from "../../assets/colors";
import ActionButton from "../ActionButton/ActionButton";

class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      password: "",
      rememberMe: true
    };
  }

  handleLogin(values) {
    // make API request to authenticate a user using the form values
  }

  getRedirectPath() {
    const locationState = this.props.location.state;

    if (locationState && locationState.from.pathname) {
      return locationState.from.pathname;
    }

    return "/";
  }

  render() {
    const authenticated = false;

    if (authenticated) {
      return (
        <Redirect
          to={{
            pathname: this.getRedirectPath(),
            state: {
              from: this.props.location
            }
          }}
        />
      );
    }
    const inputStyle = `margin: 12px 4px !important; display: block; max-width: 375px;`;
    return (
      <div>
        <MenuBar />
        <div
          css={`
            display: grid;
            grid-template-columns: 45% 55%;
            height: 100vh;
          `}
        >
          <div
            css={`
              padding: 125px 5vw 5vw;
              height: 100%;
            `}
          >
            <h1
              css={`
                color: ${colors.blue};
              `}
            >
              QHacks
            </h1>
            <p
              css={`
                margin-top: 24px;
                color: #8a929f;
              `}
            >
              Welcome back.
            </p>
            <p
              css={`
                margin-top: 12px;
                color: #8a929f;
              `}
            >
              Please login to your account.
            </p>
            <div
              css={`
                margin-top: 64px;
              `}
            >
              <input
                css={inputStyle}
                id="email"
                type="text"
                value={this.state.email}
                onChange={(e) => this.setState({ email: e.target.value })}
                placeholder="Enter your email address"
              />
              <input
                css={inputStyle}
                id="password"
                type="password"
                value={this.state.password}
                onChange={(e) => this.setState({ password: e.target.value })}
                placeholder="Enter your password"
              />
            </div>
            <div
              css={`
                margin: 40px 0;
                display: flex;
              `}
            >
              <div
                css={`
                  flex-grow: 1;
                `}
              >
                <input
                  type="checkbox"
                  id="rememberMe"
                  onChange={(e) =>
                    this.setState({ rememberMe: e.target.checked })
                  }
                />
                <label htmlFor="rememberMe">Remember me</label>
              </div>
              <div
                css={`
                  flex-grow: 1;
                `}
              >
                <Link to="/forgot-password">Forgot password</Link>
              </div>
            </div>
            <div>
              <ActionButton color="red">Login</ActionButton>
            </div>
          </div>
          <div
            css={`
              background: url(${require("../../assets/img/circuits.png")})
                no-repeat center center fixed;
              background-size: cover;
              height: 100%;
            `}
          />
        </div>
      </div>
    );
  }
}

export default Login;
