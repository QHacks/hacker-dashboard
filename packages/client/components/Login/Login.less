@columnPadding: 50px;

.login-container {
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: center;
  height: 100vh;
  width: 100vw;
}

.login-form-wrapper {
  align-items: center;
  box-shadow: 3px 0 25px rgba(0, 0, 0, 0.3);
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  padding: @columnPadding;
  width: 40%;
  z-index: 0;
}

.qhacks-logo-login {
  height: 130px;
  margin-bottom: 20px;
}

.login-picture {
  align-items: center;
  background-color: #000;
  background-image: url("../../assets/img/students-at-hackathon.jpg");
  background-position: center;
  background-size: cover;
  color: white;
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  width: 60%;
}

.login-picture-overlay {
  align-items: center;
  background-color: rgba(0, 0, 0, 0.6);
  display: flex;
  height: 100%;
  justify-content: center;
  padding: @columnPadding;
  width: 100%;
}
