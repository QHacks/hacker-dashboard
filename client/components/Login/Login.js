import { actionCreators, selectors } from '../../HackerStore';
import { Redirect } from 'react-router-dom';
import LoginForm from '../utils/LoginForm';
import { Header } from 'semantic-ui-react';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import './Login.less';

const { login } = actionCreators;
const { getLoginLoading, getAuthenticated, getIsLoginError } = selectors;

class Login extends Component {

	constructor(props) {
		super(props);

		this.renderLoginForm.bind(this);
		this.getRedirectPath.bind(this);
	}

	handleLogin(values) {
		this.props.login(values);
	}

	getRedirectPath() {
			const locationState = this.props.location.state;

			if (locationState && locationState.from.pathname) {
				return locationState.from.pathname;
			}

			return '/dashboard';
	}

	renderLoginForm() {
		const { loginLoading, loginError } = this.props;
		return (
			<div className="login-form-wrapper">
				<img src={require('../../assets/img/qhacks-tricolor-logo.svg')}
					className="qhacks-logo" />
				<LoginForm onSubmit={this.handleLogin.bind(this)}
					loginLoading={loginLoading}
					loginError={loginError} />
			</div>
		);
	}

	renderLoginDisplay() {
		return (
			<div className="login-picture">
				<div className="login-picture-overlay">
					<div>
						<Header as="h1"
							className="fontWeight-normal"
							inverted
							size="huge">
							Dream it. Build it.
						</Header>
						<Header as="h2"
							className="fontWeight-lighter"
							inverted
							size="medium">
							QHacks 2018, Queen's University
						</Header>
					</div>
				</div>
			</div>
		);
	}

	render() {
		const { authenticated } = this.props;

		if (authenticated) {
			return (
				<Redirect to={{
					pathname: this.getRedirectPath(), state: {
						from: this.props.location
					}
				}}/>
			);
		}

		return (
			<div className="login-container">
				{this.renderLoginForm()}
				{this.renderLoginDisplay()}
			</div>
		);
	}
}

function mapStateToProps(state, ownProps) {
	return {
		...ownProps,
		authenticated: getAuthenticated(state),
		loginLoading: getLoginLoading(state),
		loginError: getIsLoginError(state)
	};
}

export default connect(mapStateToProps, { login })(Login);
