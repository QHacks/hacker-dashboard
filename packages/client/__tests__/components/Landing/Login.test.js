import React from "react";
import Login from "../../../components/Landing/Login";
import { shallow } from "enzyme";

describe("Login", () => {
  it("renders correctly", () => {
    expect(shallow(<Login />)).toMatchSnapshot();
  });
});
