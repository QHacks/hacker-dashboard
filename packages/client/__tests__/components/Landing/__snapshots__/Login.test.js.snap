// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<Mutation
  ignoreResults={true}
  mutation={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "LoginUser",
          },
          "operation": "mutation",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "input",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "input",
                      },
                    },
                  },
                ],
                "directives": Array [
                  Object {
                    "arguments": Array [],
                    "kind": "Directive",
                    "name": Object {
                      "kind": "Name",
                      "value": "client",
                    },
                  },
                ],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "login",
                },
                "selectionSet": undefined,
              },
            ],
          },
          "variableDefinitions": Array [
            Object {
              "defaultValue": undefined,
              "directives": Array [],
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "LoginInput",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "input",
                },
              },
            },
          ],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 82,
        "start": 0,
      },
    }
  }
>
  <Component />
</Mutation>
`;
